rust/src/lib.rs

```diff
@@ -1,88 +1,16 @@
 //! Produce & verify TrueLayer API `Tl-Signature` request headers.
-//!
-//! # Example
-//! ```no_run
-//! # fn main() -> Result<(), truelayer_signing::Error> {
-//! # let (kid, private_key, idempotency_key, body) = unimplemented!();
-//! // `Tl-Signature` value to send with the request.
-//! let tl_signature = truelayer_signing::sign_with_pem(kid, private_key)
-//!     .method("POST")
-//!     .path("/payouts")
-//!     .header("Idempotency-Key", idempotency_key)
-//!     .body(body)
-//!     .sign()?;
-//! # Ok(()) }
-//! ```
 mod base64;
+mod common;
 mod http;
 mod jws;
 mod openssl;
 mod sign;
 mod verify;
 
+pub use http::{Get, Post};
 pub use jws::JwsHeader;
-pub use sign::Signer;
-pub use verify::Verifier;
-
-/// Start building a request `Tl-Signature` header value using private key
-/// pem data & the key's `kid`.
-///
-/// # Example
-/// ```no_run
-/// # fn main() -> Result<(), truelayer_signing::Error> {
-/// # let (kid, private_key, idempotency_key, body) = unimplemented!();
-/// let tl_signature = truelayer_signing::sign_with_pem(kid, private_key)
-///     .method("POST")
-///     .path("/payouts")
-///     .header("Idempotency-Key", idempotency_key)
-///     .body(body)
-///     .sign()?;
-/// # Ok(()) }
-/// ```
-pub fn sign_with_pem<'a>(kid: &'a str, private_key_pem: &'a [u8]) -> Signer<'a> {
-    Signer::new(kid, private_key_pem)
-}
```

Comment:
Are we completely moving away from those? It feels like this should require some alignment internally.

Reply:
Perhaps we should only introduce the new functionality in this iteration, what do you think?

Reply:
I've re-introduced the `sign_with_pen` and `verify_with..` functions. There are still some breaking changes, e.g., now you need to call the `build` functions and use the Http `Method` enum. 

The docs have been updated to reflect the changes.